api/v1beta1/properties.go

```diff
@@ -108,22 +108,22 @@ func (dk *DynaKube) ActiveGateImage() string {
 	return fmt.Sprintf("%s/linux/activegate:latest", registry)
 }
 
-func (dk *DynaKube) ServerlessMode() bool {
+func (dk *DynaKube) UseCSIDriver() bool {
 	if dk.CloudNativeFullstackMode() {
-		return dk.Spec.OneAgent.CloudNativeFullStack.ServerlessMode
+		return true
 	} else if dk.ApplicationMonitoringMode() {
-		return dk.Spec.OneAgent.ApplicationMonitoring.ServerlessMode
+		return dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver != nil && *dk.Spec.OneAgent.ApplicationMonitoring.UseCSIDriver
 	}
 	return false
 }
 
 func (dk *DynaKube) NeedsCSI() bool {
```

Comment:
Please add some unit tests for the possible combinations in `properties_test.go`